proposals/0345-android-12-issues.md

```diff
@@ -0,0 +1,322 @@
+
+This will require that the exported flag be set to true in the manifest for the apps `SdlService` as an external apps `SdlRouterService` would be the one starting the service.
+
+#### Library Changes
+##### AndroidTools.java
+~~~ java
+//Creating the PendingIntent from SdlRouterService.java
+public static void sendExplicitBroadcast(Context context, Intent intent, List<ResolveInfo> apps) {
+    //...
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && intent.getBooleanExtra(TransportConstants.PENDING_BOOLEAN_EXTRA, false)) {
+        Intent pending = new Intent();
+        PendingIntent pendingIntent = PendingIntent.getForegroundService(context, (int) System.currentTimeMillis(), pending, PendingIntent.FLAG_MUTABLE | Intent.FILL_IN_COMPONENT);
+        intent.putExtra(TransportConstants.PENDING_INTENT_EXTRA, pendingIntent);
+    }
+
+    context.sendBroadcast(intent);
+    //...
+}
+~~~
+
+#### App Developer Changes
+##### AndroidManifest.xml
+~~~ xml
+<service
+    android:name="com.sdl.hellosdlandroid.SdlService"
+    android:exported="true" <!--New Addition-->
+    android:foregroundServiceType="connectedDevice">
+</service>
+~~~
+
+##### SdlReceiver.java
+~~~ java
+//Retrieve, Update, and Send the PendingIntent
+@Override
+public void onSdlEnabled(Context context, Intent intent) {
+    DebugTool.logInfo(TAG, "SDL Enabled");
+    intent.setClass(context, SdlService.class);
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+        if (intent.getParcelableExtra(TransportConstants.PENDING_INTENT_EXTRA) != null) {
+            PendingIntent pendingIntent = (PendingIntent) intent.getParcelableExtra(TransportConstants.PENDING_INTENT_EXTRA);
+            try {
+                //Here we are allowing the RouterSerivce that is in the Foreground to start the SdlService on our behalf
+                pendingIntent.send(context, 0, intent);
+            } catch (PendingIntent.CanceledException e) {
+                e.printStackTrace();
+            }
+        }
+    } else {
+        // SdlService needs to be foregrounded in Android O and above
+        // This will prevent apps in the background from crashing when they try to start SdlService
+        // Because Android O doesn't allow background apps to start background services
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(intent);
+        } else {
+            context.startService(intent);
+        }
+    }
+}
+~~~
+
+### Bluetooth Runtime Permissions
+With the new required Bluetooth Runtime permissions we will need developers to include the new permissions in the `AndroidManifest.xml` file. 
+
+#### Developer Changes
```

Comment:
```suggestion
#### App Developer Changes
```

Does it make sense to move this section to after "Library Changes" to keep the same order as the previous section?

Reply:
My reasoning for this order is that this is directly related to the topic in the line above and it makes sense to start with the changes to the manifest before the impact the permissions has on the library.